src/plugins/gs-plugin-eos.c

```diff
@@ -106,6 +106,32 @@ on_desktop_apps_changed (GDBusConnection *connection,
 }
 
 static void
+ensure_default_branches (FlatpakInstallation *installation)
+{
+	g_autoptr(GPtrArray) xremotes = NULL;
+	guint i;
+
+	xremotes = flatpak_installation_list_remotes (installation, NULL, NULL);
+	if (xremotes == NULL || xremotes->len == 0)
+		return;
+
+	for (i = 0; i < xremotes->len; i++) {
+		FlatpakRemote *xremote = g_ptr_array_index (xremotes, i);
+		const gchar *remote_name = flatpak_remote_get_name (xremote);
+		g_autofree char *default_branch = NULL;
+		g_autoptr(GError) error_local = NULL;
+
+		default_branch = flatpak_remote_get_default_branch (xremote);
+		if (default_branch == NULL) {
```

Comment:
We usually invert the check and continue in cases like this.
Either that or join both _if_ statements in one:
`if (default_branch == NULL && !flatpak_installation.... )`